components/Graphic.jsx

```diff
@@ -279,12 +292,17 @@ var Graphic = React.createClass({
     var dataURL = this.refs.canvas.toDataURL();
     var img = new Image();
     img.src = dataURL;
+    var dpr = window.devicePixelRatio || 1;
+    img.devicePixelRatio = dpr;
     return img;
   },
 
   setPanelCount: function(c) {
+    this.panelCount = c;
     var cvs = this.refs.canvas;
-    cvs.width = c * this.defaultWidth;
+    var dpr = window.devicePixelRatio || 1;
```

Comment:
we can turn this one into just `cvs.width = c * this.defaultWidth * this.getPixelRatio();`